### PR TITLE
Test PR for review ONLY

### PR DESCRIPTION
**Pull request by rolfschr**
DON'T MERGE

This PR can be used to review all changes (compared to template project repo).

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -1,28 +1,77 @@
...
+      qrcodegen-version = getVersion qrcodegen-src;
+      androsphinx-version = getVersion androsphinx-src;
 
       # System types to support.
       supportedSystems = [ "x86_64-linux" ];
 
+      # Mapping from Nix' "system" to Android's "system".
+      androidSystemByNixSystem = {
+        "x86_64-linux" = "linux-x86_64";
+        "x86_64-darwin" = "darwin-x86_64";
```
Comment (ysndr) [inline flake.nix:77]: is this necessary at all? especially as `darwin-` is also contradictive with the name and the  only supported system is `linux`

File: flake.nix [hunk 2]
```diff
@@ -32,89 +81,173 @@
...
+            buildToolsVersions = [ "28.0.3" "29.0.3" ];
+            platformVersions = [ "29" ];
+            abiVersions = [ "x86" "x86_64" "armeabi-v7a" "armeabi-v8a" ];
+            includeNDK = true;
+            ndkVersion = "22.0.7026061";
+          };
+          ndk = sdk.ndk-bundle;
+          buildPythonPackage = python3.pkgs.buildPythonPackage;
+          zxcvbn = python3.pkgs.zxcvbn;
+          androidSystem = androidSystemByNixSystem.${system};
```
Comment (ysndr) [inline flake.nix]: is this a necessary output at all?

Comment (ysndr) [inline flake.nix]: is this a necessary output at all?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+# (Now scan the generated code with the phone.)
+```
+
+The phone is now configured correctly.
+Make sure the server process (``oracle``) is running.
+Use the app's search form to search for "site".
+You should see a log statement from the sphinx server as well as the entry "user" on the phone.
+Try copying the password to the clipboard by using the master password from above.
+You should receive the same password as the CLI client.
+
```
Comment (ysndr) [inline README.md:167]: You might want to add some update instructions, what repo to pull androsphinx and generate the gradle env from or provide an update script for future contributors.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+* [androsphinx](https://github.com/dnet/androsphinx): an Android app wrapping libsphinx.
+  * Dependencies: libsodium, libsphinx
+
+Note: The androsphinx readme suggests to use ``qrencode`` to generate a QR code that is used to configure the phone. Similarly, pwdsphinx' readme suggests ``qrcodegen``.
+These tools fulfill the same task.
+
+Strictly speaking, this Flake packages more than needed.
+The additional packages (``pwdsphinx`` & its upstream dependencies) allows to easily test the Android app.
+
+# Local test setup
```
Comment (ysndr) [inline README.md:40]: Thanks for providing these very detailed instructions

**Review by ysndr — APPROVED** [review]
Code looks good to me

**Comment by ysndr** [comment]
The Nix part and readme ade done really well!
Id only recommend an update script to update the gradle env

Again, sorry for taking so long

File: flake.nix (hunk 2 above)
Comment (rolfschr) [reply flake.nix]: No, you're right. I'll move it to where it is needed.

File: flake.nix (hunk 1 above)
Comment (rolfschr) [reply flake.nix:77]: Yes, only Linux is supported. However, I wanted to highlight the fact that some work has been done here. I'll add an explanatory comment.

**Comment by rolfschr** [comment]
Closing this as the review has been done.

**Pull request closed without merging**